testcases/OpTestEEH.py

```diff
@@ -379,3 +204,180 @@ def check_eeh_pe_recovery(self, pe):
             if pe in device:
                 return True
         return False
+
+
+class OpTestEEHbasic_fenced_phb(OpTestEEH):
+    ##
+    # @brief  This testcase has below steps
+    #         1. Get the list of pci PHB domains
+    #         2. Get the root PHB domain where the root file system
+    #            is installed(We need to skip this as EEH recovery will
+    #            fail on root PHB).
+    #         3. Start injecting the fenced PHB errors in a for loop
+    #            Only one time basic check whether PHB recovered or not
+    #
+    # @return BMC_CONST.FW_SUCCESS or BMC_CONST.FW_FAILED
+    #
+    def runTest(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.cv_SYSTEM.host_console_login()
+        self.cv_SYSTEM.host_console_unique_prompt()
+
+        root_domain = self.cv_HOST.host_get_root_phb()
+        pci_domains = self.cv_HOST.host_get_list_of_pci_domains()
+        print "Skipping the root phb %s for fenced PHB Testcase" % root_domain
+        pci_domains.remove(root_domain)
+        self.cv_HOST.host_gather_opal_msg_log()
+        self.cv_HOST.host_gather_kernel_log()
+        l_con = self.cv_SYSTEM.sys_get_ipmi_console()
+        self.cv_SYSTEM.host_console_login()
+        self.cv_SYSTEM.host_console_unique_prompt()
+        l_con.run_command("stty cols 300")
+        l_con.run_command("stty rows 10")
+        l_con.run_command("dmesg -D")
+        l_con.run_command("uname -a")
+        l_con.run_command("cat /etc/os-release")
+        for domain in pci_domains:
+            self.prepare_logs()
+            cmd = "echo 0x8000000000000000 > /sys/kernel/debug/powerpc/%s/err_injct_outbound; lspci;" % domain
+            print "=================Injecting the fenced PHB error on PHB: %s=================" % domain
+            l_con.run_command(cmd)
+            # Give some time to EEH PCI Error recovery
+            time.sleep(30)
+            l_con.run_command("\n")
```

Comment:
EEH tests are completely not running due to two issues, Remove this line.
` l_con.run_command("\n")`


And also we need fix in OpTestIPMI
diff --git a/common/OpTestIPMI.py b/common/OpTestIPMI.py
index c1c3df3..b9ff4fd 100644
--- a/common/OpTestIPMI.py
+++ b/common/OpTestIPMI.py
@@ -154,7 +154,7 @@ class IPMIConsole():
         console = self.get_console()
         console.sendline(command)
         console.expect("\n") # from us
-        rc = console.expect_exact("[console-pexpect]#", timeout)
+        rc = console.expect("\[console-pexpect\]#$", timeout)
 
         if rc == 0:
             res = console.before
@@ -162,7 +162,7 @@ class IPMIConsole():
             return res
         else:
             res = console.before
-            res = res.split(i_cmd)
+            res = res.split(command)
             return res[-1].splitlines()`
`